pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -616,9 +616,50 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 		changes.KillPod = true
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) {
+		if pod.Spec.RestartPolicy != v1.RestartPolicyAlways && changes.KillPod == false {
```

Comment:
in practice, usually containers don't return 0 when gets killed proactively. for this impl, after we kill sidecar, and pod restart policy is OnFailure, previous logic will restart sidecar containers, and could possibly cause infinite loop here.

Reply:
yeah, that's right, I hadn't considered that, I'll see what I can do to get around that

Reply:
and also, given containers don't return 0 most of time when killed, might want to consider re-implement pod phase, else good pod can reach Failed phase simply because sidecar gets killed by kubelet.

https://github.com/zhan849/kubernetes/commit/11909879517b91b2252e39823333f2a053fd2d89 FYI

Reply:
I've implemented a change to make sure we don't get a restart loop in this case. I agree it is a bit awkward in regards to Pod phase as this pod would be considered Running in this case. I think this needs more debate though as changing how pod phase works could cause unexpected problems in other places.